docs/how-to-migrate-from-smartlock.md

```diff
@@ -0,0 +1,791 @@
+Google Smart Lock for Passwords to the OpenYOLO API. If your application does
+not currently utilize Smart Lock for Passwords, we recommend instead reading
+the main [README](../README.md), which contains a getting started guide and
+links to other resources.
+
+Google will continue to offer the Smart Lock for Passwords API for the
+foreseeable future. In 2018, the Smart Lock for Passwords API will start
+automatically using the OpenYOLO protocol, however we encourage developers to
+explicitly move to the OpenYOLO API, as this is where all future
+work and innovation will occur.
+
+OpenYOLO for Android is intended to be the open-standards based successor to
+the Google Smart Lock for Passwords API. OpenYOLO provides the same
+functionality and UX flows as Google Smart Lock for Passwords, while also
+allowing the user to choose from any compatible credential manager.
+
+Migration is a straightforward process, as the APIs are structurally very
+similar. This document will demonstrate how to migrate to the OpenYOLO API for
+each API method that is offered in the Smart Lock for Passwords API.
+
+## Importing the OpenYOLO API
+
+To use the OpenYOLO API, first add the API as a dependency on your project. If
+you do not use any of the other Google authentication features (like Google
+Sign-in), you can also remove the dependency on Play Services Authentication:
+
+```diff
+dependencies {
+-  // optionally, remove play services auth if you no longer require it
+-  implementation 'com.google.android.gms:play-services-auth:x.y.z'
++  // add the OpenYOLO API
++  implementation 'org.openyolo:openyolo-api:x.y.z'
+}
+```
+
+Unlike Smart Lock for Passwords, the OpenYOLO client library will work on
+devices without Google Play Services.
+
+## Account hints
+
+Smart Lock for Passwords defined the concept of a hint request, which allows
+an app to solicit a credential to simplify new account creation or existing
+account discovery. Hints provide a user identifier and authentication method,
+and may optionally include a display name, profile picture and generated
+password.
+
+### Sending a hint request
+
+In both the Smart Lock for Passwords API and the OpenYOLO API, Intents are used
+to communicate with the user’s credential provider to retrieve an account hint.
+
+With Smart Lock for Passwords, a hint request for an email, phone number or
+Google Sign-in based account would be constructed and dispatched as follows:
+
+```java
+void sendSmartLockHintRequest() {
+  PendingIntent hintPickerIntent =
+      Auth.CredentialsApi.getHintPickerIntent(
+          mGoogleApiClient,
+          new HintRequest.Builder()
+              .setEmailAddressIdentifierSupported(true)
+              .setPhoneNumberIdentifierSupported(true)
+              .setAccountTypes(IdentityProviders.GOOGLE)
+              .setIdTokenRequested(true)
+              .setServerClientId(MY_GOOGLE_CLIENT_ID)
+              .setIdTokenNonce(“nonceValue”)
+              .set
```

Comment:
Looks like this line is incomplete

Reply:
Removed, I think was walking through all our methods to demonstrate them and had been copy-pasting the prefix.